### PR TITLE
feat(ui): All jobs only accessible by admin

### LINKED ISSUE #244: Changing the UI so that only Admin can access the `AllRecentJobs` page
### Description

As the APIs for jobs are ready and also there is a API for sending all jobs which can only be accessed by the Admin. 
(see here: https://github.com/fossology/fossology/pull/2260)
We can make the `AllRecentJobs` page accessible to the only if the logged in user is Admin. Otherwise the route in the Navigation bar can be hidden for all other users. 

PS: I would like to work on this issue

@GMishx @shaheemazmalmmd @Shruti3004

### PR DESCRIPTION
**Pull request by soham4abc**
## Description

- All recent jobs is now only accessible  by admins ( Related with this back-end PR:  https://github.com/fossology/fossology/pull/2260 This needs to be merged first in order to avoid bugs.)
- MyRecentJobs page also implemented.

### Changes
Introduced a new API endpoint and used conditions for the header.

## How to test

Run compile the back-end after PR: ( https://github.com/fossology/fossology/pull/2260 ) is merged and run the server. Use Admin and non-Admin accounts to test out the jobs pages.

## Video 

[Screencast from 18-07-22 01:23:54 PM IST.webm](https://user-images.githubusercontent.com/63705023/179469079-a1990390-aaf9-4958-a2d5-e34bb14271c6.webm)

Fixes #244

PTAL @GMishx @shaheemazmalmmd @Shruti3004

### CODE REVIEW

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Comment by soham4abc** [comment]
Does this require any more changes? @GMishx

**Comment by soham4abc** [comment]
@GMishx

**Comment by soham4abc** [comment]
@Shruti3004

File: src/api/jobs.js [hunk 1]
```diff
@@ -36,6 +36,20 @@ export const getAllJobApi = ({ page, limit }) => {
     },
   });
 };
+
```
Comment (Shruti3004) [inline src/api/jobs.js:39]: Please add one liner about your function as documentation

File: src/pages/Jobs/MyRecentJobs/index.jsx [hunk 2]
```diff
@@ -16,20 +17,210 @@
...
+      type: "success",
+      text: messages.loading,
+    });
+    setShowMessage(true);
+    getAllJob(jobsData)
+      .then((res) => {
+        // Formatting the date from time stamp to readable date
+        setShowMessage(false);
+        const arr = [];
+        for (let i = 0; i < res.res.length; i++) {
```
Comment (Shruti3004) [inline src/pages/Jobs/MyRecentJobs/index.jsx:72]: Please add condtions there like: 
`res?.res?.length`

**Review by Shruti3004 — CHANGES_REQUESTED** [review]

File: src/services/jobs.js [hunk 3]
```diff
@@ -47,6 +48,15 @@ export const getAllJob = (jobsDatalist) => {
   });
 };
```
Comment (Shruti3004) [inline src/services/jobs.js:50]: One liner here as well

File: src/api/jobs.js (hunk 1 above)
Comment (soham4abc) [reply src/api/jobs.js:39]: resolved

File: src/pages/Jobs/MyRecentJobs/index.jsx (hunk 2 above)
Comment (soham4abc) [reply src/pages/Jobs/MyRecentJobs/index.jsx]: resolved

File: src/services/jobs.js (hunk 3 above)
Comment (soham4abc) [reply src/services/jobs.js:50]: resolved

**Comment by soham4abc** [comment]
@Shruti3004 Does this need any more changes??

**Pull request merged**